src/cpu/dyn_cache.h

```diff
@@ -738,21 +788,32 @@ static void cache_init(bool enable) {
 				cache_blocks[i].cache.next = &cache_blocks[i + 1];
 			}
 		}
-		if (cache_code_start_ptr==NULL) {
+		if (cache_code_start_ptr==nullptr) {
 			// allocate the code cache memory
 #if defined (WIN32)
-			cache_code_start_ptr=(Bit8u*)VirtualAlloc(0,CACHE_TOTAL+CACHE_MAXSIZE+PAGESIZE_TEMP-1+PAGESIZE_TEMP,
-				MEM_COMMIT,PAGE_EXECUTE_READWRITE);
+			cache_code_start_ptr=static_cast<uint8_t *>(VirtualAlloc(0,cache_code_size,
+				MEM_COMMIT,PAGE_EXECUTE_READWRITE));
 			if (!cache_code_start_ptr)
-				cache_code_start_ptr=(Bit8u*)malloc(CACHE_TOTAL+CACHE_MAXSIZE+PAGESIZE_TEMP-1+PAGESIZE_TEMP);
+				cache_code_start_ptr=static_cast<uint8_t *>(malloc(cache_code_size));
```

Comment:
Add brackets to this if, (because the prior if needs them.  Once a parent or child needs them, we use them for all parents or children to help readability)